references/segmentation/train.py

```diff
@@ -11,15 +11,18 @@
 from coco_utils import get_coco
 from torch import nn
 from torch.optim.lr_scheduler import PolynomialLR
-from torchvision.transforms import functional as F, InterpolationMode
+from torchvision.prototype.transforms import Compose, functional as F, InterpolationMode
 
 
 def get_dataset(dir_path, name, image_set, transform):
-    def sbd(*args, **kwargs):
-        return torchvision.datasets.SBDataset(*args, mode="segmentation", **kwargs)
+    def voc(*args, transforms, **kwargs):
+        return torchvision.datasets.VOCSegmentation(*args, transforms=Compose([transforms]), **kwargs)
```

Comment:
*This comment references some other parts of this PR. I'll leave some pointers below.*

---

This `Compose` looks redundant, but it isn't. The old, i.e. custom segmentation compose, was implemented like

```py
def forward(image, target):
    for transform in self.transforms:
        image, target = transform(image, target)
```

while the new roughly does

```py
def forward(sample):
    for transform in self.transforms:
        sample = transform(sample)
```

This in itself is not an issue for the new transforms since `transform(image, target)` and `transform((image, target))` are equivalent. However, if we run with `--weights` and `--test-only`, we define a `preprocessing` function that acts like a transformation. This function cannot deal with both types of inputs and thus we have to align the outputs of the different datasets.

Since COCO uses a `Compose` internally, I've opted to also use one here.